app/controllers/contacts_controller.rb

```diff
@@ -0,0 +1,68 @@
+class ContactsController < ApplicationController
+  before_action :find_contact, only: %i[edit update show destory]
+  before_action :find_multiple_contacts, only: %i[activate deactivate]
+
+  def index
+    @contacts = Contact.all
+  end
+
+  def show; end
+
+  def edit; end
+
+  def update
+    @contact.update(contact_params)
+
+    redirect_back fallback_location: root_path
+  end
+
+  def activate
+    update_status(true)
+
+    render_contact_row
+  end
+
+  def deactivate
+    update_status(false)
+
+    render_contact_row
+  end
+
+  private
+
+  def contact_params
+    params.require(:contact).permit(:first_name, :last_name, :email, :active)
+  end
+
+  def find_contact
+    @contact = Contact.find(params[:id])
+  end
+
+  def find_multiple_contacts
+    @contacts = Contact.where(id: params[:ids])
+  end
+
+  def update_status(status)
```

Comment:
```suggestion
  def update_status(status:)
```
nit: I like to always use keyword/named arguments for booleans